lib/ometajs/core.js

```diff
@@ -365,10 +369,27 @@
 			this._getBranches = function() {
 				return branches;
 			};
+			this._not = function(x) {
+				var self = this,
+					origInput = this.input,
+					origAddBranch = this._addBranch,
```

Comment:
Why do we have to store/restore the current `_addBranch` and `_addToken`?

Reply:
So this is the same as the previous code (it's just previously it was unconditional whereas now it's only when `_addBranch`/`_addToken` will be something other than a `noop`), but the reasoning is because `_not` is when we're specifically looking to *not* match, and `_addBranch` is used for adding auto-completion hints and `_addToken` is used for highlighting.  And we don't want to add suggestions for things that specifically must not match, or highlight things that are invalid (because the lack of highlighting is useful for knowing that something is not valid in that space)